File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int n = 0;
        int r;
        int temp = x;
        
        while (x > 0) {
            r = x % 10;
            n = n * 10 + r;
            x = x/10;
        }
        
        if(temp == n) {
            return true;
        } 
        else {
            return false;
        }
        
    }
    
    // public static void main(String[] args) {
    //     Solution s = new Solution();
    //     s.isPalindrome(121);
    // }
}

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int temp = x;
        unsigned int n = 0;
        int r;
        
        while(x > 0){
            r = x%10;
            n = n*10+r;
            x = x/10;
        }
        
        if(temp == n){
            return true;
        }
        else{
            return false;
        }
    }
};